### ISSUE TITLE
Debugger should support symbol and source server

### ISSUE THREAD

**Issue by gregg-miskelly** [OmniSharp/omnisharp-vscode#373]
In Visual Studio, it is possible for the debugger to automatically download symbols and sources for the dlls being debugged (as long as the dll/exe in question is built using Windows PDBs).

The Roslyn team is planning to support both of these features for portable PDBs. This issue tracks consuming this feature from VS Code once they are available.

**Comment by 0xdeafcafe** [comment]
Any word on this?

**Comment by gregg-miskelly** [comment]
For source server, the debugger has been working on [/sourcelink](https://github.com/dotnet/roslyn/issues/12759) and we hope to ship this in one of the upcoming releases.

Folks are still working on coming up with a full plan symbol server and I don't know the latest. As a stop gap, the Roslyn team added support for [/debug:embedded](https://github.com/dotnet/roslyn/issues/12390) and the currently shipping version of the C# extension supports this. While this isn't a good option for all scenarios, depending on what your scenario is, it might make sense.

**Comment by ctaggart** [comment]
Any updated status on this?

**Comment by psibernetic** [comment]
Checking in on this, planning on enabling SourceLink in my projects for my customers, but I primarily work in VS Code these days, would like to have the same capability.

**Comment by gregg-miskelly** [comment]
This isn't something that we have started on yet. We have started the development work for authentication support for SourceLink in VS. That is the last major solution location initiative for VS. So once that is done I would hope we would have the cycles to come back and get at least unauthenticated source link + embedded source working in VS Code and VS For Mac. 

Authentication + Source Link + VS Code / VS For Mac would be tricky. So that will almost certainly be further out than the first two.

**Comment by gregg-miskelly** [comment]
This feature is now complete and I released a beta (1.15-beta1) with the feature. See [instructions](https://github.com/OmniSharp/omnisharp-vscode/wiki/Installing-Beta-Releases) to install.

Servers that require authentication are NOT currently supported. I opened a new issue to track that: https://github.com/OmniSharp/omnisharp-vscode/issues/2071

Feel free to '+1' on that if your scenario requires authentication.
